### PR TITLE
Feature/flow 2129

### PR DESCRIPTION
**Pull request by jamesbrattBoomi**

### CODE REVIEW

File: js/services/validation.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import * as Model from './model';
 import * as Settings from './settings';
 import * as Utils from './utils';
 
-const isValueDefined = function (value: any, contentType: string) {
+const isValueNotDefined = function (value: any, contentType: string) {
```
Comment (dshoulders) [inline js/services/validation.ts:6]: This doesn't seem right to me. Having the function check for a "not" positive is confusing.

File: js/services/validation.ts [hunk 2]
```diff
@@ -123,11 +123,13 @@ export const validate = (model: any, state: any, flowKey: string): IValidationRe
...
 export const validateString = (value: string, regex: string | null, message: string, isRequired: boolean, flowKey: string): IValidationResult => {
-    if (isRequired && isValueDefined(value, Component.contentTypes.string)) {
+    if (isRequired && isValueNotDefined(value, Component.contentTypes.string)) {
         return getRequiredResponse(message, flowKey);
     }
 
-    if (!validateRegex(value, regex)) {
+    // Only optional fields get here...
+    // But the regex check should only trigger a validation error if a value has been set
+    if (!validateRegex(value, regex) && !isValueNotDefined(value, Component.contentTypes.string)) {
```
Comment (dshoulders) [inline js/services/validation.ts:132]: This gets more confusing when isValueNotDefined is inverted. Is this value not not defined?

File: js/services/validation.ts (hunk 1 above)
Comment (jamesbrattBoomi) [reply js/services/validation.ts]: Sorted

File: js/services/validation.ts (hunk 2 above)
Comment (jamesbrattBoomi) [reply js/services/validation.ts]: Sorted

**Pull request merged**